src/main/java/com/cyberark/conjur/api/clients/AuthnClient.java

```diff
@@ -45,7 +45,9 @@ public AuthnClient(final Credentials credentials,
 
         // replacing the password with an API key
         this.apiKey = credentials.getPassword();
-        if(credentials.getAuthnUrl().endsWith("/authn")){
+        // Only for the below cases password is used and login() is required:
+        // authentication method is API-key based and a non-host entity is authenticating
```

Comment:
This might be a bit clearer
```suggestion
        // login() exchanges a user's password for an API key. This is not relevant for hosts since they
        // don't have passwords (only API keys). It's also only relevant for standard API key authentication
        // (authn), not other authentication methods.
```